### PR TITLE
(towards #1383) Standardise Directive node_str and remove unneeded code

### PR DESCRIPTION
**Pull request by sergisiso**

### CODE REVIEW

**Comment by sergisiso** [comment]
@rupertford @arporter This is ready for review. It does not close the issue because I want to follow up with a similar clean up of the dag_name method. Btw `grep -r "def node_str" -A 15` can be used to see that all remaining node_str are now doing some extra from the generic node_str.

File: src/psyclone/tests/psyir/nodes/acc_directives_test.py [hunk 1]
```diff
@@ -68,41 +68,28 @@ def setup():
     Config._instance = None
```
Comment (arporter) [inline src/psyclone/tests/psyir/nodes/acc_directives_test.py:70]: As per our discussion in the meeting on Monday, please could you precede this test with a comment saying which class method it is for (see later in this file for examples). In fact, we also need a comment to say which class the following tests are for (i.e. ACCLoopDirective). Please could you ensure that this test is in the right location in this file - i.e. that it follows the ordering of the classes in the `acc_directives.py` file.

**Review by arporter — CHANGES_REQUESTED** [review]
Looks good Sergi, thanks for improving things.
All tests and examples are OK, with compilation and with and without termcolor.
All changes are pylint/pycodestyle clean.
Ref and Dev guides build fine and do not require updating.
Just a few minor typos to fix.

File: src/psyclone/tests/nemo/nemo_psy_test.py [hunk 2]
```diff
@@ -234,6 +234,7 @@ def test_schedule_view(capsys):
         "UNDEFINED>]\n" +
         6*indent + sched_str + "[]\n" +
         7*indent + "0: " + kern_str + "[]\n")
+    print(output)
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply src/psyclone/tests/nemo/nemo_psy_test.py:237]: Done

File: src/psyclone/tests/psyir/nodes/acc_directives_test.py [hunk 3]
```diff
@@ -68,41 +68,28 @@ def setup():
...
+    assert directive.node_str() == expected
+    assert str(directive) == expected
 
-    # Loop directive with collapse
-    acclt.apply(schedule[1].dir_body[0].dir_body[0], {"collapse": 2})
-    out = schedule[1].dir_body[0].dir_body[0].node_str()
-    assert out.startswith(
-        colored("Directive", colour) + "[ACC Loop, collapse=2, "
-                                       "independent]")
+    # Non-default value outptu
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply src/psyclone/tests/psyir/nodes/acc_directives_test.py:85]: Done

File: src/psyclone/tests/psyir/nodes/node_test.py [hunk 4]
```diff
@@ -107,22 +111,21 @@ def dummy(_1, _2):
...
-
-    # Manually set the _text_name and _colour for this node to
-    # something that will result in coloured output (if requested
-    # *and* termcolor is installed).
-    tnode._text_name = "FakeName"
-    tnode._colour = "green"
-    assert tnode.node_str(False) == "FakeName[]"
-    assert tnode.node_str(True) == colored("FakeName", "green") + "[]"
+
+    # Mock the coloured_name method because it is already testes elsewhere
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply src/psyclone/tests/psyir/nodes/node_test.py:118]: Done

File: src/psyclone/tests/psyir/nodes/acc_directives_test.py (hunk 1 above)
Comment (sergisiso) [reply src/psyclone/tests/psyir/nodes/acc_directives_test.py:70]: Done. I reordered all test classes that weren't on the right location. This certainly exposes where we have coverage but we are not really unit testing as some classes are even missing from this file.

**Comment by sergisiso** [comment]
@arporter This is ready for another look

File: src/psyclone/tests/psyir/nodes/acc_directives_test.py (hunk 1 above)
Comment (arporter) [reply src/psyclone/tests/psyir/nodes/acc_directives_test.py:70]: Thanks Sergi.

**Review by arporter — APPROVED** [review]
All requested changes have been made.
All tests and examples still OK.
Dev and Ref guides build with no new warnings.
Changes still pylint clean.

**Pull request merged**